set.mm

```diff
@@ -261350,11 +261373,41 @@ irreducible elements (see ~ df-irred ).  (Contributed by Mario Carneiro,
       ABESBGSUTVAAWGOVBVCVDVFVGVHVTVMVPVMVSTABVNVOCWAWBVIVJVKVL $.
   $}
 
+  ${
+    $d B x y $.  $d L x y $.  $d K x y $.  $d ph x y $.
+    nzrpropd.1 $e |- ( ph -> B = ( Base ` K ) ) $.
+    nzrpropd.2 $e |- ( ph -> B = ( Base ` L ) ) $.
+    nzrpropd.3 $e |- ( ( ph /\ ( x e. B /\ y e. B ) ) -> ( x ( +g ` K ) y ) =
+      ( x ( +g ` L ) y ) ) $.
+    nzrpropd.4 $e |- ( ( ph /\ ( x e. B /\ y e. B ) ) -> ( x ( .r ` K ) y ) =
+      ( x ( .r ` L ) y ) ) $.
+    $( If two structures have the same components (properties), one is a
+       nonzero ring iff the other one is.  (Contributed by SN, 21-Jun-2025.) $)
+    nzrpropd $p |- ( ph -> ( K e. NzRing <-> L e. NzRing ) ) $=
+      ( crg wcel cur cfv c0g wne wa cnzr eqid isnzr rngidpropd neeq12d anbi12d
+      ringpropd grpidpropd 3bitr4g ) AEKLZEMNZEONZPZQFKLZFMNZFONZPZQERLFRLAUGUK
+      UJUNABCDEFGHIJUDAUHULUIUMABCDEFGHJUAABCDEFGHIUEUBUCEUHUIUHSUISTFULUMULSUM
+      STUF $.
+  $}
+
   ${
     opprnzr.1 $e |- O = ( oppR ` R ) $.
+    $( The opposite of a nonzero ring is nonzero, bidirectional form of
+       ~ opprnzr .  (Contributed by SN, 20-Jun-2025.) $)
+    opprnzrb $p |- ( R e. NzRing <-> O e. NzRing ) $=
+      ( crg wcel cur cfv c0g wne cnzr opprringb anbi1i eqid isnzr oppr1 3bitr4i
+      wa oppr0 ) ADEZAFGZAHGZIZQBDEZUBQAJEBJESUCUBABCKLATUATMZUAMZNBTUAATBCUDOA
+      BUACUERNP $.
+
     $( The opposite of a nonzero ring is nonzero.  (Contributed by Mario
-       Carneiro, 17-Jun-2015.) $)
+       Carneiro, 17-Jun-2015.)  (Proof shortened by SN, 20-Jun-2025.) $)
     opprnzr $p |- ( R e. NzRing -> O e. NzRing ) $=
+      ( cnzr wcel opprnzrb biimpi ) ADEBDEABCFG $.
+
+    $( Obsolete version of ~ opprnzr as of 20-Jun-2025.  (Contributed by Mario
```

Comment:
opprnzrb is a new theorem, and as a fallout it may replace proof steps in other theorems.  opprnzr is one direction of opprnzb, so its shortening is very obvious, perhaps even automatically available, and not at all surprising.  I lean in such cases towards seeing those shortenings as minimization rather than new proof with extra characteristics.  Then we neither need an OLD version nor a shortening tag, see [conventions](https://us.metamath.org/mpeuni/conventions.html):

> When a proof is shortened automatically (using
       the Metamath program "MM-PA> MINIMIZE__WITH *" command), then it is not
       necessary to keep the old proof, nor to add credit for the shortening.
       </li>